CONFIG.md

```diff
@@ -0,0 +1,151 @@
+# config
+
+The canonical documentation of shpool's config is the comments
+on the `Config` struct defined in `libshpool/src/config.rs`, but
+this document aims to provide some high level explanations of
+some common configuration options.
+
+You can specify the path to your config file by passing a
+`-c /path/to/config.toml` flag, or by creating and
+editing `~/.config/shpool/config.toml`.
+
+## Prompt Prefix
+
+By default, `shpool` will detect when you are using a shell it knows
+how to inject a prompt into. Currently, those shells include `bash`,
+`zsh` and `fish`, but more may be added in the future. If it noticed
+you are using one such shell, it will inject the prompt prefix
+`shpool:$SHPOOL_SESSION_NAME` at the beginning of your prompt
+in order to hint to you when you are inside of a `shpool` session.
+
+You can customize this prompt prefix by setting a new value in
+your config. For example, to show the `shpool` session name
+inside square brackets, you can put
+
+```
+prompt_prefix = "[$SHPOOL_SESSION_NAME]"
+```
+
+in your config file. If you want to instead completely suppress
+the prompt injection, you can just set a blank `prompt_prefix`
+with
+
+```
+prompt_prefix = ""
+```
+
+this allows you to write a custom prompt hook in your .rc files
+that examines the `$SHPOOL_SESSION_NAME` environment variable
+directly, or eschew a `shpool` prompt customization entirely.
+
+## Session Restore Mode
+
+`shpool` can do a few different things when you re-attach to an existing
+session. You can choose what you want it to do with the `session_restore_mode`
+configuration option.
+
+### `"screen"` (default) - restore a screenful of history
+
+The `"screen"` option causes `shpool` to re-draw sufficient output to fill the
+entire screen of the client terminal as well as using the SIGWINCH trick
+described in the `"simple"` section below. This will help restore
```

Comment:
```suggestion
described in the `"simple"` section below. This will help to restore
```


```suggestion
described in the `"simple"` section below. This will help restoring
```

Choose you team